### DISCUSSION TITLE
Tibber vs photovoltaics

### DISCUSSION THREAD

**Discussion by Hooch76** [AmyJeanes/TeslaMateAgile#87] [General]
How does TeslaMateAgile handle the costs if the car is charged with solar power (and no energy is used from the grid / Tibber)? This obiously happens when at home (geofence). 
I do not see a parameter like "energy consumed from grid" when looking at all the config options so my guess is that TeslaMateAgile will read the energy added to the Tesla from the Teslamate database assuming that all was taken from the grid. True?

Thanks

**Comment by AmyJeanes** [comment]
Yes that's right, all energy will be assumed to be from the grid. In my personal scenario, I do have battery and solar but use Octopus Intelligent which always charges overnight from the cleaner / cheaper grid.

If you'd like to take account of solar panels / local generation / storage etc in the price calculation I'd recommend using the Home Assistant provider instead, there are so many types of solar panels and inverter types it would be impossible for me to integrate cleanly with them all.

To do this, use Home Assistant to constantly maintain a number/input_number entity which contains the current price of electricity, including taking into account local generation and/or battery storage. There are tons of integrations in HA to help you to do this and also tons for all the smart energy providers such as Tibber out there as well.

You can then plug this entity into TeslaMateAgile and it'll use that entity to calculate pricing for a charge after it's completed, just make sure the recorder is enabled for that entity as it needs to read the history for it (it will be by default).

Hope that helps!
